processing/src/main/java/io/druid/segment/NullHandlingHelper.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import io.druid.query.aggregation.AggregateCombiner;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.BufferAggregator;
+import io.druid.query.aggregation.NullableAggregateCombiner;
+import io.druid.query.aggregation.NullableAggregator;
+import io.druid.query.aggregation.NullableBufferAggregator;
+
+public class NullHandlingHelper
+{
+  // use these values to ensure that convertObjectToLong(), convertObjectToDouble() and convertObjectToFloat()
+  // return the same boxed object when returning a constant zero.
+  public static final Double ZERO_DOUBLE = 0.0d;
+  public static final Float ZERO_FLOAT = 0.0f;
+  public static final Long ZERO_LONG = 0L;
+
+  // Using static injection to avoid adding JacksonInject annotations all over the code.
+  @Inject
+  private static NullValueHandlingConfig INSTANCE = new NullValueHandlingConfig(true);
+
+  public static boolean useDefaultValuesForNull()
```

Comment:
What if someone wants to override the behavior for a specific query? It will be tough to migrate if this is an all or nothing switch -- behavior will change suddenly.

Also, will things work if some historicals have one setting for null handling behavior and some have a different one? This must work properly or else rolling config changes will be broken.